egcg_core/rest_communication.py

```diff
@@ -70,10 +70,9 @@ def _req(self, method, url, quiet=False, **kwargs):
         if r.status_code in self.successful_statuses:
             if not quiet:
                 self.debug(report)
-        elif r.status_code == 401:
-            raise RestCommunicationError('Invalid auth credentials')
         else:
             self.error(report)
+            raise RestCommunicationError('Encountered a %s status code: %s' % (r.status_code, r.reason))
```

Comment:
Do we want to raise exceptions like this? Previous behaviour was just to log errors. Alternatively, we could raise exceptions only on certain status codes (401 auth required, 500 internal server error, etc)